sdk/src/main/java/io/dapr/actors/runtime/ActorTypeInformation.java

```diff
@@ -0,0 +1,181 @@
+     */
+    private final boolean remindable;
+
+    /**
+     * Instantiates a new {@link ActorTypeInformation}
+     * @param name Actor type's name.
+     * @param implementationClass Actor's implementation class.
+     * @param interfaces Actor's immediate interfaces.
+     * @param abstractClass Whether Actor type is abstract.
+     * @param remindable Whether Actor type is remindable.
+     */
+    private ActorTypeInformation(String name,
+                                 Class implementationClass,
+                                 Collection<Class> interfaces,
+                                 boolean abstractClass,
+                                 boolean remindable) {
+        this.name = name;
+        this.implementationClass = implementationClass;
+        this.interfaces = interfaces;
+        this.abstractClass = abstractClass;
+        this.remindable = remindable;
+    }
+
+    /**
+     * Returns the name of this ActorType.
+     * @return ActorType's name.
+     */
+    public String getName() {
+        return this.name;
+    }
+
+    /**
+     * Gets the type of the class implementing the actor.
+     *
+     * @return The {@link Class} of implementing the actor.
+     */
+    public Class getImplementationClass() {
+        return this.implementationClass;
+    }
+
+    /**
+     * Gets the actor interfaces which derive from {@link Actor} and implemented by actor class.
+     *
+     * @return Collection of actor interfaces.
+     */
+    public Collection<Class> getInterfaces() {
+        return Collections.unmodifiableCollection(this.interfaces);
+    }
+
+    /**
+     * Gets a value indicating whether the class implementing actor is abstract.
+     *
+     * @return true if the class implementing actor is abstract, otherwise false.
+     */
+    public boolean isAbstractClass() {
+        return this.abstractClass;
+    }
+
+    /**
+     * Gets a value indicating whether the actor class implements {@link Remindable}.
+     *
+     * @return true if the actor class implements {@link Remindable}.
+     */
+    public boolean isRemindable() {
+        return this.remindable;
+    }
+
+    /**
+     * Creates the {@link ActorTypeInformation} from given Class.
+     *
+     * @param actorClass The type of class implementing the actor to create ActorTypeInformation for.
+     * @return ActorTypeInformation if successfully created for actorType or null.
+     */
+    public static ActorTypeInformation tryParse(Class actorClass) {
+        try {
+            return parse(actorClass);
+        } catch (IllegalArgumentException e) {
+            return null;
+        }
+    }
+
+    /**
+     * Creates an {@link #ActorTypeInformation} from actorType.
+     *
+     * @param actorClass The class implementing the actor to create ActorTypeInformation for.
+     * @return {@link #ActorTypeInformation} created from actorType.
+     */
+    public static ActorTypeInformation parse(Class actorClass) {

```

Comment:
Agree this is better than "Get()".  This looks like a factory.  What do you think about construct/build/create instead?

Reply:
I have considered the 4 options on the table. Let me know which one you feel stronger about.

Why parse?
It can throw an exception, so we have a tryParse as well that can return null. This is why parse came to me naturally: we see parse() and tryParse() everywhere. Also, the logic to construct this object is very heavy, so parse represents that effort IMO.

Why not construct?
I am not used this the "construct" term in Java code. It is just my experience.

Why not build?
It gives the idea that it will never fail and always return a valid result.

Why maybe create?
I have seen the term create before for things like this. I am fine with this option as well.

Reply:
Renamed to create()